tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -448,6 +448,19 @@ void StorageManagerCanonical::delete_array(const char* array_name) {
   vfs()->remove_files(compute_tp(), array_dir.array_meta_uris());
   vfs()->remove_files(compute_tp(), array_dir.fragment_meta_uris());
   vfs()->remove_files(compute_tp(), array_dir.array_schema_uris());
+
+  // Delete all tiledb child directories
+  // Note: using vfs()->ls() here could delete user data
+  std::vector<URI> dirs;
```

Comment:
Lets make a vector in constants.cc that contains all array directories located right next to the declarations of array_commits_dir_name, array_fragment_meta_dir_name, etc. This code should use this vector to generate the dirs. That way if someone add a directory later, they have more chances of noticing the vector and adding to it, which will mean this code will continue to work.

Reply:
Same comment for the group directories.